staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -334,13 +335,18 @@ func strategicPatchObject(
 func (p *patcher) applyPatch(_ context.Context, _, currentObject runtime.Object) (objToUpdate runtime.Object, patchErr error) {
 	// Make sure we actually have a persisted currentObject
 	p.trace.Step("About to apply patch")
-	if hasUID, err := hasUID(currentObject); err != nil {
+	if currentObjectHasUID, err := hasUID(currentObject); err != nil {
 		return nil, err
-	} else if !hasUID {
+	} else if currentObjectHasUID {
+		objToUpdate, patchErr = p.mechanism.applyPatchToCurrentObject(currentObject)
+	} else {
 		// TODO: Check with the authorizer if the user has permission to create
 		objToUpdate, patchErr = p.mechanism.createNewObject()
-	} else {
-		objToUpdate, patchErr = p.mechanism.applyPatchToCurrentObject(currentObject)
+		if objToUpdateHasUID, err := hasUID(objToUpdate); err != nil {
```

Comment:
is it OK to apply with the UID?

Reply:
uid is a precondition that requires we match the existing object... it's the only way you can guarantee your applied patch is merged into the same object. if you don't want that precondition, don't include uid

Reply:
OK, So it's safe to assume that the code will check that. Great, thanks :-)